### ISSUE TITLE
Support for active geo-replication in `azurerm_redis_enterprise_database`

### ISSUE THREAD

**Issue by MrDrMcCoy** [hashicorp/terraform-provider-azurerm#16448]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add support for [active geo-replication](https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-how-to-active-geo-replication) in the Enterprise Azure Cache for Redis. The only other way to automate this is [via Bicep](https://docs.microsoft.com/en-us/azure/templates/microsoft.cache/redisenterprise/databases?tabs=bicep#databaseproperties), which we do not wish to introduce into our Terraform environment.

### New or Affected Resource(s)/Data Source(s)

azurerm_redis_enterprise_database

### Potential Terraform Configuration

```hcl
resource "azurerm_redis_enterprise_database" "example" {
  name                = "default"
  resource_group_name = azurerm_resource_group.example.name
  cluster_id = azurerm_redis_enterprise_cluster.example.id
  geo_replication {
    name = "groupNickname"
    linked_databases = ["id", ...]
  }
}
```

### References

Docs:

- https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-how-to-active-geo-replication
- https://docs.microsoft.com/en-us/azure/templates/microsoft.cache/redisenterprise/databases?tabs=bicep#databaseproperties

**Comment by xiaxyi** [comment]
The feature is merged:https://github.com/hashicorp/terraform-provider-azurerm/pull/16045

**Comment by MrDrMcCoy** [comment]
@xiaxyi This is very exciting, but I am not able to get it to work. Given the following resources:

```hcl
resource "azurerm_redis_enterprise_cluster" "geo" {
  for_each = var.config.aks_regions
  location = each.key
  minimum_tls_version = "1.2"
  name = "anark${var.resource_group.name}${each.key}geo"
  resource_group_name = var.resource_group.name
  sku_name = var.config.redis_geo.sku_name
}

resource "azurerm_redis_enterprise_database" "geo" {
  name = "default" # we have to use this name as of azurerm 3.3.0 2022/04/22
  client_protocol = "Encrypted"
  cluster_id = azurerm_redis_enterprise_cluster.geo[keys(var.config.aks_regions)[0]].id
  clustering_policy = "EnterpriseCluster"
  linked_database_id = [
    for _, value in azurerm_redis_enterprise_cluster.geo :
      "${value.id}/databases/default"
  ]
}
```

Which results in the following error:

```
│ Error: evictionPolicy must be set to NoEviction when using RediSearch module
│ 
│   with module.az_resource_group["dev2"].azurerm_redis_enterprise_database.geo,
│   on az_resource_group/redis.tf line 37, in resource "azurerm_redis_enterprise_database" "geo":
│   37: resource "azurerm_redis_enterprise_database" "geo" {
```

We are not interested in the `RediSearch` module, nor are we interested in disabling eviction. From the available documentation that I have seen, there is no way to exclude these add-on modules. Should I open a new issue for this?

**Comment by xiaxyi** [comment]
@MrDrMcCoy Let me check the code, besides, can you be more specific about the `there is no way to exclude these add-on modules`?  

no need to open another issue, we can do the follow-up in the thread.

**Comment by MrDrMcCoy** [comment]
> can you be more specific about the `there is no way to exclude these add-on modules`?

Sure. The `module {}` block only allows you to specify `name` and `args`, but not `enabled`. The fact that I am getting the error above despite not requesting any modules to be installed suggests that these modules are getting enabled by default when creating a geo-replicated database. Short of a mechanism to exclude these modules, the default behavior of this resource creation would need to be changed to not include them.

**Comment by xiaxyi** [comment]
@MrDrMcCoy During the creation of the database, if the module is not included in the config, then no module will be created by default.

can you try below TF config and see if the database can be linked?
```
resource "azurerm_redis_enterprise_cluster" "test" {
  name                = "test-rec"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_resource_group.test.location

  sku_name = "Enterprise_E20-4"
}
resource "azurerm_redis_enterprise_cluster" "test1" {
  name                = "test-rec1"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_resource_group.test.location

  sku_name = "Enterprise_E20-4"
}

resource "azurerm_redis_enterprise_database" "test" {
  cluster_id          = azurerm_redis_enterprise_cluster.test.id
  resource_group_name = azurerm_resource_group.test.name

  client_protocol   = "Encrypted"
  clustering_policy = "EnterpriseCluster"
  eviction_policy   = "AllKeysLRU"
  linked_database_id = [
    "${azurerm_redis_enterprise_cluster.test.id}/databases/default",
    "${azurerm_redis_enterprise_cluster.test1.id}/databases/default",
  ]

 linked_database_group_nickname = "tftestGeoGroup"
}
```

**Comment by xiaxyi** [comment]
@MrDrMcCoy May I know if any of the databases which are involved in the active-geo linking process are created before?

**Comment by xiaxyi** [comment]
@MrDrMcCoy yes, Thanks for the clarification. I will perform more tests to have the issue reproduced from my side

**Comment by xiaxyi** [comment]
https://github.com/hashicorp/terraform-provider-azurerm/pull/16700

**Comment by katbyte** [comment]
It appears this issue relates to `azurerm_redis_enterprise_database`, which was removed in v5.0 in favour of `azurerm_managed_redis_database` (see the [v5.0 upgrade guide](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/5.0-upgrade-guide#azurerm_redis_enterprise_database) for details), as such we are going to close this issue out as it no longer applies to the current version of the provider.

If this is still relevant on the latest version of the provider (v5.x) please open a new issue against `azurerm_managed_redis_database` referencing this one. Thank you!
